staging/src/k8s.io/client-go/tools/metrics/metrics.go

```diff
@@ -43,15 +43,24 @@ var (
 	RequestResult ResultMetric = noopResult{}
 )
 
-// Register registers metrics for the rest client to use. This can
-// only be called once.
+// Register registers given latency and result metrics for the rest client to use.
+// This should only be called once.
 func Register(lm LatencyMetric, rm ResultMetric) {
 	registerMetrics.Do(func() {
 		RequestLatency = lm
 		RequestResult = rm
 	})
 }
 
+// RegisterNew registers new latency and result metrics for the rest client to use.
+// This should only be called once.
+func RegisterNew() {
```

Comment:
Or we can initialize both metrics via `Register()` at `request.go`, but I think `RegisterNew` is a better abstraction.

Reply:
registerMetrics makes Register and RegisterNew mutually exclusive. That seems non-obvious and likely to confuse things. At a minimum we should be documenting that you should only be calling one of Register and RegisterNew once. I think that is more confusing that just having just Register.

Reply:
Yes, after careful thought, I think adding `RegisterNew` makes things much complicated. I will modify `Register` in such manner that we don't need to initialize both the metrics at `request.go`.